data_import/signal_helpers.py

```diff
@@ -5,6 +5,45 @@
 from .models import DataRetrievalTask
 
 
+def task_signal_pre_save(task_params, datafile_model, **kwargs):
+    """
+    Trigger data retrieval a study adds new data via UserData's data field.
+
+    Data retrieval task creation is triggered on a pre_save signal to a
+    UserData object (derived from studies.models.BaseStudyUserData) and only
+    occurs if the new instance's data field is non-empty and changed.
+
+    If the user's email address is verified, the task is started. Otherwise it
+    is postponed.
+    """
+    instance = kwargs['instance']
+
+    # Require the object looks like a UserData-derived object
+    if not hasattr(instance, 'data') and hasattr(instance, 'user'):
+        return
+
+    # Only create a task if the new 'data' field is not empty and has changed.
+    if not instance.data:
+        return
+    curr_version = kwargs['sender'].objects.get(pk=instance.pk)
```

Comment:
@beaugunderson This seems like an unlikely scenario... but another thought also occurred to me – what if UserData is newly created and unsaved (but will be) _and_ the 'data' field is not empty? In which case the correct behavior would be to go ahead (trigger a task). I've added a check on the pre_save instance's pk for this scenario.